Pokemon app sarvajeet

dheerajshrivastva-dev opened the pull request.

Sanujkumar committed:
initial files added

sarvajeet12 committed:
initial commit

sarvajeet12 committed:
Delete README.md

sarvajeet12 committed:
Rename readme-file.md to README.md

Sanujkumar committed:
Resolved merge conflict with origin/main

Sanujkumar committed:
Merge pull request #1 from Bevarc-Intern-Batch-Dec-2024-25/pokeman_app_sanuj

Pokeman app sanuj

sarvajeet12 committed:
Update README.md

sarvajeet12 committed:
Merge branch 'main' into pokemon-app-sarvajeet

dheerajshrivastva-dev commented on src/components/show.tsx:
```diff
@@ -0,0 +1,95 @@
+  const navigate = useNavigate();
+
+  const fetchPokemons = async (page: number) => {
+    try {
+      setLoading(true);
+      const offset = (page - 1) * ITEMS_PER_PAGE;
+      const response = await axios.get(
+        `https://pokeapi.co/api/v2/pokemon?offset=${offset}&limit=${ITEMS_PER_PAGE}`
+      );
+      setPokemonList(response.data.results);
+      setTotalPages(Math.ceil(response.data.count / ITEMS_PER_PAGE));
+      setLoading(false);
+    } catch (err) {
+      console.error(err);
+      setLoading(false);
+    }
+  };
+
```
@sarvajeet12 Why are we using axios alone to fetch the data? I think you did not through with instructions in group. Please use react query.

sarvajeet12 commented:
This is not my code sir, this is Sanuj Kumar Code. I am using React query
sir ..

On Thu, Jan 2, 2025, 11:48 Dheeraj Shrivastva ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/components/show.tsx
> <https://github.com/Bevarc-Intern-Batch-Dec-2024-25/Pokemon-App/pull/6#discussion_r1900568645>
> :
>
> > +    try {
> +      setLoading(true);
> +      const offset = (page - 1) * ITEMS_PER_PAGE;
> +      const response = await axios.get(
> +        `https://pokeapi.co/api/v2/pokemon?offset=${offset}&limit=${ITEMS_PER_PAGE}` <https://pokeapi.co/api/v2/pokemon?offset=$%7Boffset%7D&limit=$%7BITEMS_PER_PAGE%7D>
> +      );
> +      setPokemonList(response.data.results);
> +      setTotalPages(Math.ceil(response.data.count / ITEMS_PER_PAGE));
> +      setLoading(false);
> +    } catch (err) {
> +      console.error(err);
> +      setLoading(false);
> +    }
> +  };
> +
>
> @sarvajeet12 <https://github.com/sarvajeet12> Why are we using axios
> alone to fetch the data? I think you did not through with instructions in
> group. Please use react query.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Bevarc-Intern-Batch-Dec-2024-25/Pokemon-App/pull/6#pullrequestreview-2527065717>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7SL2OUQBWTLVBHVEJTPCQL2ITK33AVCNFSM6AAAAABUPJ75MGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKMRXGA3DKNZRG4>
> .
> You are receiving this because you were mentioned.Message ID:
> <Bevarc-Intern-Batch-Dec-2024-25/Pokemon-App/pull/6/review/2527065717@
> github.com>
>

sarvajeet12 committed:
Delete vite.config.ts

sarvajeet12 committed:
Delete tsconfig.app.json

sarvajeet12 committed:
Delete tsconfig.node.json

sarvajeet12 committed:
Delete tsconfig.json

sarvajeet12 committed:
Delete src/vite-env.d.ts

sarvajeet12 committed:
Delete src/main.tsx

sarvajeet12 committed:
Delete src/App.tsx

sarvajeet12 committed:
Delete src/App.css

sarvajeet12 committed:
Delete src/components/pokemanDetails.tsx

sarvajeet12 committed:
Delete src/components/show.tsx

sarvajeet12 commented:
Due to merge issue sir this file come in my branch sir,...i fix this issue
. Now sir you can check my branch

On Thu, Jan 2, 2025, 12:31 SARVAJEET SHAH ***@***.***>
wrote:

> This is not my code sir, this is Sanuj Kumar Code. I am using React query
> sir ..
>
> On Thu, Jan 2, 2025, 11:48 Dheeraj Shrivastva ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In src/components/show.tsx
>> <https://github.com/Bevarc-Intern-Batch-Dec-2024-25/Pokemon-App/pull/6#discussion_r1900568645>
>> :
>>
>> > +    try {
>> +      setLoading(true);
>> +      const offset = (page - 1) * ITEMS_PER_PAGE;
>> +      const response = await axios.get(
>> +        `https://pokeapi.co/api/v2/pokemon?offset=${offset}&limit=${ITEMS_PER_PAGE}` <https://pokeapi.co/api/v2/pokemon?offset=$%7Boffset%7D&limit=$%7BITEMS_PER_PAGE%7D>
>> +      );
>> +      setPokemonList(response.data.results);
>> +      setTotalPages(Math.ceil(response.data.count / ITEMS_PER_PAGE));
>> +      setLoading(false);
>> +    } catch (err) {
>> +      console.error(err);
>> +      setLoading(false);
>> +    }
>> +  };
>> +
>>
>> @sarvajeet12 <https://github.com/sarvajeet12> Why are we using axios
>> alone to fetch the data? I think you did not through with instructions in
>> group. Please use react query.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Bevarc-Intern-Batch-Dec-2024-25/Pokemon-App/pull/6#pullrequestreview-2527065717>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A7SL2OUQBWTLVBHVEJTPCQL2ITK33AVCNFSM6AAAAABUPJ75MGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKMRXGA3DKNZRG4>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> <Bevarc-Intern-Batch-Dec-2024-25/Pokemon-App/pull/6/review/2527065717@
>> github.com>
>>
>

sarvajeet12 committed:
Update README.md 

When this error give : 'vite' is not recognized as an internal or external command, operable program or batch file.

Then use this command to resolve : 
npm install vite --save-dev

sarvajeet12 committed:
Update README.md

sarvajeet12 committed:
Update README.md

dheerajshrivastva-dev commented:
Code is written with passion :+1:
